openeogeotrellis/integrations/keycloak.py

```diff
@@ -0,0 +1,25 @@
+import logging
+
+import requests
+
+# TODO: point this to prod
+KEYCLOAK = "https://sso-int.terrascope.be"
+
+_log = logging.getLogger(__name__)
+
+
+def authenticate_oidc(client_id: str, client_secret: str, logging_context=None) -> str:
+    if logging_context is None:
+        logging_context = {}
+
+    with requests.post(f"{KEYCLOAK}/auth/realms/terrascope/protocol/openid-connect/token", data={
```

Comment:
FYI:  the new CDAS project will introduce some non-terrascope keycloak usage, so hardlinking "keycloak" with the terrascope instance might not scale well.

Ususally it's best to start (config-wise) from an OIDC "issuer" URL, `https://sso-int.terrascope.be/auth/realms/terrascope` in this case, append `/.well-known/openid-configuration` to it and discover the token URL (and other capabilities) from that

Reply:
also FYI: the openeo client lib already implements this client credentials logic:
```python
import openeo.rest.auth.oidc

oidc_provider = openeo.rest.auth.oidc.OidcProviderInfo(
    issuer="https://sso-int.terrascope.be/auth/realms/terrascope"
)
client_info = openeo.rest.auth.oidc.OidcClientInfo(
    provider=oidc_provider,
    client_id=...,
    client_secret=...,
)   
authenticator = openeo.rest.auth.oidc.OidcClientCredentialsAuthenticator(
    client_info=client_info
)
access_token = authenticator.get_tokens().access_token
```